pkg/storage/redisProvider.go

```diff
@@ -73,29 +85,54 @@ func (provider *Redis) Name() string {
 
 // ListKeys method returns the list of existing keys
 func (provider *Redis) ListKeys() []string {
-	keys, _ := provider.inClient.Do(provider.ctx, provider.inClient.B().Keys().Pattern("*").Build()).AsStrSlice()
+	provider.logger.Sugar().Debugf("Call the ListKeys function in redis")
+	var scan redis.ScanEntry
+	var err error
+	elements := []string{}
+	for more := true; more; more = scan.Cursor != 0 {
+		if scan, err = provider.inClient.Do(context.Background(), provider.inClient.B().Scan().Cursor(scan.Cursor).Match(provider.hashtags+MappingKeyPrefix+"*").Build()).AsScanEntry(); err != nil {
+			provider.logger.Sugar().Errorf("Cannot scan: %v", err)
+		}
+		for _, element := range scan.Elements {
+			value := provider.Get(element)
+			mapping, err := decodeMapping(value)
+			if err == nil {
+				for _, v := range mapping.Mapping {
+					if !v.FreshTime.Before(time.Now()) || !v.StaleTime.Before(time.Now()) {
+						elements = append(elements, v.RealKey)
+					}
+				}
+			}
+		}
```

Comment:
I think this is easier to read.
```suggestion
		 for _, element := range scan.Elements {
			value := provider.Get(element)
			mapping, err := decodeMapping(value)
			if err != nil {
				continue
			}
			for _, v := range mapping.Mapping {
				if v.FreshTime.Before(time.Now()) && v.StaleTime.Before(time.Now()) {
					continue
				}
				 elements = append(elements, v.RealKey)
			}
		}
```